packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -499,11 +541,15 @@ class _SegmentedControlState<T extends Object> extends State<CupertinoSlidingSeg
   // Otherwise the thumb can be dragged around in an ongoing drag gesture.
   bool? _startedOnSelectedSegment;
 
+  // Whether the current drag gesture started on a disabled segment. When this
+  // flag is true, the `onDown` and `onUpdate` methods will return directly.
```

Comment:
uber nit: prefer documenting the behavior (i.e., drag gestures will be ignored).